File: effecientcal.java
import java.util.Scanner;

public class effecientcal {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        
        System.out.print("Enter first number: ");
        int num1 = input.nextInt();
        System.out.print("Enter second number: ");
        int num2 = input.nextInt();
        
        System.out.println("Select operation: +  -  *  /");
        String op = input.next();
        
        switch (op) {
            case "+":
                System.out.println("Sum = " + (num1 + num2));
                break;
            case "-":
                System.out.println("Difference = " + (num1 - num2));
                break;
            case "*":
                System.out.println("Product = " + (num1 * num2));
                break;
            case "/":
                if (num2 == 0)
                    System.out.println("Cannot divide by zero");
                else
                    System.out.println("Quotient = " + (num1 / (double) num2));
                break;
            default:
                System.out.println("Invalid operation");
        }
        input.close();
    }
}

File: effecientcal.py
num = "nill"
num2 = "nill"
def add(num, num2):
    return num + num2
def subtract(num, num2):
    return num - num2
def multiply(num, num2):
    return num * num2
def divide(num, num2):
    return num / num2
def main():
    num = int(input("Enter a number: "))
    num2 = int(input("Enter another number: "))
    operation = input("Enter an operation (+, -, *, /): ")
    if operation == "+":
        print(add(num, num2))
    elif operation == "-":
        print(subtract(num, num2))
    elif operation == "*":
        print(multiply(num, num2))
    elif operation == "/":
        if num2 == 0:
            print("Cannot divide by zero")
            return
        print(divide(num, num2))
    else:
        print("Invalid operation")
main()